pkg/workspaces/workspaces.go

```diff
@@ -0,0 +1,224 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workspaces
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	nameParam       = "name"
+	claimNameParam  = "claimName"
+	subPathParam    = "subPath"
+	emptyDirParam   = "emptyDir"
+	configParam     = "config"
+	secretParam     = "secret"
+	configItemParam = "item"
+)
+
+const invalidWorkspace = "invalid input format for workspace parameter: "
+
+var errNotFoundParam = errors.New("param not found")
+
+// MergeWorkspace merges workspacebinding already in pipelineruns with given options
+func MergeWorkspace(ws []v1alpha1.WorkspaceBinding, optWS []string) ([]v1alpha1.WorkspaceBinding,
+	error) {
+	workspaces, err := parseWorkspace(optWS)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(workspaces) == 0 {
+		return ws, nil
+	}
+
+	for i := range ws {
+		if v, ok := workspaces[ws[i].Name]; ok {
+			ws[i] = v
+			delete(workspaces, v.Name)
+		}
+	}
+
+	for _, v := range workspaces {
```

Comment:
isn't this similar to ws = append(ws, workspaces...) ?

Reply:
Nope. It's map and slice.